test/e2e/network/network_policy.go

```diff
@@ -532,12 +640,601 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+											"ns-name": nsBName,
+										},
+									},
+									PodSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											"pod-name": nsBpodServerA.ObjectMeta.Labels["pod-name"],
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			policyAllowToServerInNSB, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowToServerInNSB)
+			framework.ExpectNoError(err, "Error occurred while creating policy: policyAllowToServerInNSB.")
+			defer cleanupNetworkPolicy(f, policyAllowToServerInNSB)
+
+			ginkgo.By("Creating client-a, in 'namespace-a', which should be able to contact the server-a in namespace-b.", func() {
+				testCanConnect(f, nsA, "client-a", nsBserviceA, 80)
+			})
+			ginkgo.By("Creating client-a, in 'namespace-a', which should not be able to contact the server-b in namespace-b.", func() {
+				testCannotConnect(f, nsA, "client-a", nsBserviceB, 80)
+			})
+			ginkgo.By("Creating client-a, in 'namespace-a', which should not be able to contact the server in namespace-a.", func() {
+				testCannotConnect(f, nsA, "client-a", service, 80)
+			})
+		})
+
+		ginkgo.It("should enforce multiple ingress policies with ingress allow-all policy taking precedence based on PodSelector [Feature:NetworkPolicy]", func() {
+			ginkgo.By("Creating a network policy for the server which allows traffic only from client-b.")
+			policyAllowOnlyFromClientB := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "allow-from-client-b-pod-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the Server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": podServerLabelSelector,
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					// Allow traffic only from "client-b"
+					Ingress: []networkingv1.NetworkPolicyIngressRule{{
+						From: []networkingv1.NetworkPolicyPeer{{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"pod-name": "client-b",
+								},
+							},
+						}},
+					}},
+				},
+			}
+
+			policyAllowOnlyFromClientB, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowOnlyFromClientB)
+			framework.ExpectNoError(err, "Error occurred while creating policy: policyAllowOnlyFromClientB.")
+			defer cleanupNetworkPolicy(f, policyAllowOnlyFromClientB)
+
+			// client-a should not be able to contact server as the policy is to allow only client-b
+			ginkgo.By("Creating client-a which should not be able to contact the server.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", service, 80)
+			})
+			// client-b should be able to contact server as the policy is to allow all
+			ginkgo.By("Creating client-b which should be able to contact the server.", func() {
```

Comment:
(and this comment is actually incorrect; the "allow all" policy doesn't exist yet)

Reply:
Removed the comment.